src/main/java/com/jvm_bloggers/core/social/fb/publisher/PageAccessTokenProvider.java

```diff
@@ -0,0 +1,90 @@
+package com.jvm_bloggers.core.social.fb.publisher;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.restfb.DefaultFacebookClient;
+import com.restfb.Version;
+import com.restfb.WebRequestor;
+
+import javaslang.control.Try;
+
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+import static com.jvm_bloggers.ApplicationProfiles.PRODUCTION;
+import static lombok.AccessLevel.PRIVATE;
+
+@Component
+@Slf4j
+@Profile({PRODUCTION})
+class PageAccessTokenProvider {
+
+    private static final String
+        PAGE_TOKEN_URL =
+        "https://graph.facebook.com/%s?fields=access_token&access_token=%s&appsecret_proof=%s";
+
+    private final DefaultFacebookClient facebookClient;
+    private final String pageId;
+    private final String appSecret;
+    private final String userAccessToken;
+    private final ObjectMapper objectMapper;
+
+    @Autowired
+    PageAccessTokenProvider(Environment env) {
+        facebookClient = new DefaultFacebookClient(Version.LATEST);
+        pageId = env.getProperty("fb.page.id");
```

Comment:
In case of multiple configuration properties related to single `module` i'd suggest to avoid multiple `@Value` and use `@ConfigurationProperties` as descibed at [Externalized Configuration](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html#boot-features-external-config-typesafe-configuration-properties)